`azurerm_cdn_endpoint_custom_domain` - add supports for HTTPS (#13283)
hashicorp/terraform-provider-azurerm: website/docs/r/cdn_endpoint_custom_domain.html.markdown

* `host_name` - (Required) The host name of the custom domain. Changing this forces a new CDN Endpoint Custom Domain to be created.

> * `cdn_managed_https` (Optional) - A `cdn_managed_https` block as defined below. 

Author: Indeed. While from my experience, the current convention is to add a separate note callout to mention the relationship between properties:

```shell
💤 grep -r "Only one of" -B 5 .
./synapse_role_assignment.html.markdown-
./synapse_role_assignment.html.markdown-* `synapse_workspace_id` - (Optional) The Synapse Workspace which the Synapse Role Assignment applies to. Changing this forces a new resource to be created.
./synapse_role_assignment.html.markdown-
./synapse_role_assignment.html.markdown-* `synapse_spark_pool_id` - (Optional) The Synapse Spark Pool which the Synapse Role Assignment applies to. Changing this forces a new resource to be created.
./synapse_role_assignment.html.markdown-
./synapse_role_assignment.html.markdown:-> **NOTE:** A Synapse firewall rule including local IP is needed to allow access. Only one of `synapse_workspace_id`, `synapse_spark_pool_id` must be set.
--
./cdn_endpoint_custom_domain.html.markdown-
./cdn_endpoint_custom_domain.html.markdown-* `cdn_managed_https` (Optional) - A `cdn_managed_https` block as defined below.
./cdn_endpoint_custom_domain.html.markdown-
./cdn_endpoint_custom_domain.html.markdown-* `user_managed_https` (Optional) - A `user_managed_https` block as defined below.
./cdn_endpoint_custom_domain.html.markdown-
./cdn_endpoint_custom_domain.html.markdown:~> **NOTE** Only one of `cdn_managed_https` and `user_managed_https` can be specified.
--
./data_factory_integration_runtime_azure_ssis.html.markdown-
./data_factory_integration_runtime_azure_ssis.html.markdown-* `subnet_name` - (Optional) Name of the subnet to which the nodes of the Azure-SSIS Integration Runtime will be added.
./data_factory_integration_runtime_azure_ssis.html.markdown-
./data_factory_integration_runtime_azure_ssis.html.markdown-* `subnet_id` - (Optional) id of the subnet to which the nodes of the Azure-SSIS Integration Runtime will be added.
./data_factory_integration_runtime_azure_ssis.html.markdown-
./data_factory_integration_runtime_azure_ssis.html.markdown:-> **NOTE** Only one of `subnet_id` and `subnet_name` can be specified. If `subnet_name` is specified, `vnet_id` must be provided.
--
./hpc_cache.html.markdown-
./hpc_cache.html.markdown-* `directory_flat_file` - (Optional) A `directory_flat_file` block as defined below.
./hpc_cache.html.markdown-
./hpc_cache.html.markdown-* `directory_ldap` - (Optional) A `directory_ldap` block as defined below.
./hpc_cache.html.markdown-
./hpc_cache.html.markdown:~> **Note:** Only one of `directory_active_directory`, `directory_flat_file` and `directory_ldap` can be set.
--
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown-* `resource_group_name` - (Required) The name of the Resource Group in which the API Management Service exists. Changing this forces a new resource to be created.
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown-* `product_id` - (Optional) The ID of the Product which should be assigned to this Subscription. Changing this forces a new resource to be created.
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown:-> **Info:** Only one of `product_id` and `api_id` can be set. If both are missing `all_apis` scope is used for the subscription.
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown-* `user_id` - (Optional) The ID of the User which should be assigned to this Subscription. Changing this forces a new resource to be created.
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown-* `api_id` - (Optional) The ID of the API which should be assigned to this Subscription. Changing this forces a new resource to be created.
./api_management_subscription.html.markdown-
./api_management_subscription.html.markdown:-> **Info:** Only one of `product_id` and `api_id` can be set. If both are missing `/apis` scope is used for the subscription and all apis are accessible.
```

> * `cdn_managed_https` (Optional) - A `cdn_managed_https` block as defined below. 

Reviewer: yeah +1 to leaving this as a separate `Note` - the big yellow/blue box is easier for users to spot imo

* `user_managed_https` (Optional) - A `user_managed_https` block as defined below.

~> **NOTE** Only one of `cdn_managed_https` and `user_managed_https` can be specified.

> !> **Warning** Due to the way the Azure API handles in-place updates on HTTPS settings for the Custom Domain of a CDN Endpoint, it is only possible to toggle these settings between disabled and enabled individually.

Reviewer: this sounds like a provider issue that we should be handling through a multi-step update?

> !> **Warning** It is allowed to update the HTTPS settings on the CDN Endpoint Custom Domain only by toggling it. In-place update on HTTPS settings is not allowed. This is because setting different HTTPS settings will need a disable-then-enable process. When HTTPS settings got disabled, the service will take ~8 hours to clean up your previous enablement request for the same custom domain and there is no way to get notification when that clean up has done.

Reviewer 2: Although the detailed explanation here is useful, I think that this level of detail belongs as a comment in the code (perhaps near the switch block in update) and would shorten this to:
```suggestion
!> **Warning** Due to the way the Azure API handles in-place updates on HTTPS settings for the Custom Domain of a CDN Endpoint, it is only possible to toggle these settings between disabled and enabled individually.
```

---

A `cdn_managed_https` block supports the following:

> * `certificate_type` - (Required) The type of the HTTPS certificate. Possible values are `Shared` and `Dedicated`.

Reviewer 2: ```suggestion
* `certificate_type` - (Required) The type of HTTPS certificate. Possible values are `Shared` and `Dedicated`.
```

* 

> * `protocol_type` - (Required) The type of the protocol. Possible values are `ServerNameIndication` and `IPBased`.

Reviewer 2: ```suggestion
* `protocol_type` - (Required) The type of protocol. Possible values are `ServerNameIndication` and `IPBased`.
```

---

A `user_managed_https` block supports the following:

* `key_vault_id` - (Required) The ID of the Key Vault that contains the HTTPS certificate.

> * `secret_name` - (Required) The name of Key Vault Certificate that contains the HTTPS certificate.

Reviewer 2: ```suggestion
* `secret_name` - (Required) The name of the Key Vault Certificate that contains the HTTPS certificate.
```

> * `secret_version` - (Required) The version of Key Vault Certificate that contains the HTTPS certificate.

Reviewer 2: ```suggestion
* `secret_version` - (Required) The version of the Key Vault Certificate that contains the HTTPS certificate.
```

The `timeouts` block allows you to
specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

> * `create` - (Defaults to 20 hours) Used when creating the CDN Endpoint Custom Domain.

Reviewer 2: Can we change all of these back to:
```suggestion
* `create` - (Defaults to 20 hours) Used when creating the Custom Domain for this CDN Endpoint.
```

Author: This part is auto generated via the website-scaffolding tool.

Reviewer 2: That doesn't mean that the phrasing is always comprehensible. This was also already [flagged in a previous PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/11120#discussion_r608729614).
